controllers/cluster_controller.go

```diff
@@ -178,6 +178,17 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, log logr.Logger
 		return ctrl.Result{}, errors.New("deleting self-managed clusters is not supported")
 	}
 
+	if metav1.HasAnnotation(cluster.ObjectMeta, anywherev1.ManagedByCLIAnnotation) {
+		log.Info("Clusters is managed by CLI, removing finalizer")
+		controllerutil.RemoveFinalizer(cluster, ClusterFinalizerName)
+		return ctrl.Result{}, nil
+	}
+
+	if cluster.IsReconcilePaused() {
+		log.Info("Cluster reconciliation is paused, won't process cluster deletion")
+		return ctrl.Result{}, nil
+	}
```

Comment:
Why is this check needed again after adding and potentially removing the finalizer?